x/ecocredit/server/marketplace/utils.go

```diff
@@ -150,6 +142,16 @@ func (k Keeper) fillOrder(ctx context.Context, sellOrder *api.SellOrder, buyerAc
 		}
 		buyerBal.Retired = retiredBalance.String()
 
+		supplyTradable, err := math.NewDecFromString(supply.TradableAmount)
+		if err != nil {
+			return err
+		}
+		supplyTradable, err = math.SafeSubBalance(supplyTradable, purchaseQty)
+		if err != nil {
+			return err
+		}
+		supply.TradableAmount = supplyTradable.String()
+
```

Comment:
this fix was applied by only editing supply in the event of retiring credits, as we need to move credits from the tradable supply to the retired supply.

Reply:
Thanks for pointing out.